# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/typedarray/dom-view.js
# Commit: a24fb00bab8e
# Full Hash: a24fb00bab8ea918e6751ddde6d7b71c12563d29
# Author: Steve Fink <sfink@mozilla.com>
# Date: 2021-10-22 03:08:27
# Description:
#   Bug 1731039 - ArrayBufferOrView subclasses should null-check in unwrap() r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D129092
# ==============================================================================

diff -r d6f67ce93bf2 -r a24fb00bab8e js/src/jit-test/tests/typedarray/dom-view.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/typedarray/dom-view.js	Thu Oct 21 17:55:30 2021 +0000
@@ -0,0 +1,13 @@
+// Bug 1731039 fuzzbug. Exercises one of the new ArrayBufferOrView classes.
+
+var error;
+try {
+  encodeAsUtf8InBuffer("", "");
+} catch (e) {
+  error = e;
+}
+
+assertEq(error.message.includes("must be a Uint8Array"), true);
+
+if (typeof reportCompare === "function")
+    reportCompare(true, true);